src/OrbitBase/include/OrbitBase/ParameterPackTrait.h

```diff
@@ -0,0 +1,203 @@
+    return {};
+  }
+
+  // Constructs a new trait object with a different variadic value type.
+  //
+  // For example that allows you to go from a trait that describes `std::tuple<int, char>` to a
+  // trait that describes `std::variant<int, char>`.
+  template <template <typename...> typename NewVariadicHolder>
+  [[nodiscard]] constexpr static ParameterPackTrait<NewVariadicHolder, Types...>
+  ChangeVariadicHolder() {
+    return {};
+  }
+
+  // Returns the number of types in the parameter pack.
+  [[nodiscard]] constexpr static size_t Size() { return sizeof...(Types); }
+
+  // Returns true iff there are duplicate types in the parameter pack. Note that this function calls
+  // `RemoveDuplicateTypes` in its implementation and the same warning about extended compile times
+  // applies here as well.
+  [[nodiscard]] constexpr static bool HasDuplicates() {
+    return RemoveDuplicateTypes().Size() < Size();
+  }
+
+  // These constexpr comparison operators allow to compare constexpr values instead of types which
+  // leads to more idiomatic unit testing code.
+  template <template <typename...> typename Other, typename... OtherTypes>
+  [[nodiscard]] friend constexpr bool operator==(
+      const ParameterPackTrait& /* lhs */,
+      const ParameterPackTrait<Other, OtherTypes...> /* rhs */) {
+    // Two ParameterPackTraits are considered equal if their types match.
+    return std::is_same_v<ParameterPackTrait, ParameterPackTrait<Other, OtherTypes...>>;
+  }
+
+  template <template <typename...> typename Other, typename... OtherTypes>
+  [[nodiscard]] friend constexpr bool operator!=(
+      const ParameterPackTrait& lhs, const ParameterPackTrait<Other, OtherTypes...> rhs) {
+    return !(lhs == rhs);
+  }
+};
+
+// MakeParameterPackTrait creates a trait object - deduced from a value.
+//
+// - We support full deduction. In that case the value's type must be an instantiation of a variadic
+//   template, in other words an instantiation of `Holder<Types...>`.
+// - We also support partial deduction. In that case the Holder template must be provided explicitly
+//   and the `Types` pack will be deduced from the value. If the value's type is an instantiation of
+//   `Holder<Others...>` then `Others...` will become the pack. If the value's type is NOT an
+//   instantiation of `Holder`, then the pack will consist of a single type `decltype(value)`.
+//   Consider the following examples:
+//   `MakePa...(variant<int, char>{})` returns `ParameterPackTrait<variant, int, char>{}`.
+//   `MakePa...<variant>(variant<int, char>{})` returns `ParameterPackTrait<variant, int, char>{}`.
+//   `MakePa...<variant>(0.0f)` returns `ParameterPackTrait<variant, float>{}`.
+//
+//   This is useful in particular for generic code. It allows to construct a type that's always an
+//   instance of `Holder` while avoiding double wrapping (`Holder<Holder<..>>`).
+template <template <typename...> typename VariadicHolder, typename... Types>
```

Comment:
Not sure about the two last functions either. Why not just using clauses?

Reply:
Changed